### PR TITLE
Added relevant files from core.ai NodeJS template 

### PR DESCRIPTION
**Pull request by Ushmajit**
Fixed minor comments from the previous PR and copied files from NodeJS template to support the following:

Template link: https://github.com/aicore/template-nodejs

1. Mocha integration for running Tests.
2. Coverage and reporting.
3. Github actions.

### CODE REVIEW

**Comment by Ushmajit** [comment]
Created a new tracking issue to fix the ES lint errors https://github.com/aicore/linode-object-storage-lib/issues/4 

Will fix this in the upcoming commits

**Comment by abose** [comment]
Thanks for the merge :)

File: README.md [hunk 1]
```diff
@@ -29,4 +29,127 @@ await linodeModule.uploadFileToLinodeBucket(accessKeyId,
 const fileURL = await linodeModule.fetchObjectUrl(accessToken, 
     clusterId, bucketName, objectName);
 ```
-    
+
+## Code Guardian
+[![<app> build verification](https://github.com/aicore/linode-object-storage-lib/actions/workflows/build_verify.yml/badge.svg)](https://github.com/aicore/linode-object-storage-lib/actions/workflows/build_verify.yml)
+
+<a href="https://sonarcloud.io/summary/new_code?id=aicore_template-nodejs-ts">
```
Comment (abose) [inline README.md:36]: Need changing to `id=aicore_linode-object-storage-lib` from `id=aicore_template-nodejs-ts` and similar below

Comment (abose) [reply README.md:36]: Also code guardian section is best laid out on top of readme to get quick statuses of the repo

Comment (Ushmajit) [reply README.md:36]: @abose Agreed, will refactor the README based on the comments above.

**Pull request merged**